package.json

```diff
@@ -1,22 +1,50 @@
 {
-  "name": "post-message-stream",
+  "name": "@metamask/post-message-stream",
   "version": "4.0.0",
   "description": "Sets up a duplex object stream over window.postMessage",
-  "main": "index.js",
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/",
+    "access": "public"
+  },
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist/"
+  ],
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "yarn build && yarn testOnly",
+    "testOnly": "jest --runInBand",
+    "test:watch": "jest --watch",
+    "prepublishOnly": "npm run build",
+    "lint": "eslint --ext .ts,.js ./src",
+    "build": "yarn run build:clean && yarn run build:compile && yarn run build:bundle",
+    "build:clean": "rm -rf ./dist && mkdir -p ./dist/esm && mkdir -p ./dist/cjs",
+    "build:compile": "yarn build:compile:esm && yarn build:compile:cjs",
+    "build:compile:esm": "tsc --module ES6 --outDir ./dist/esm",
+    "build:compile:cjs": "tsc --module UMD --outDir ./dist/cjs",
```

Comment:
Looking at this locally, I'm not sure I follow what having two builds is for? Could we ship only the CommonJS version?

I'm also confused by the naming here, though maybe it's a typo: `build:compile:cjs` is outputting UMD to the `/cjs`? Should that be outputting CJS instead?

We could move these into `tsconfig(.*).json` files, e.g.:

```diff
@@ -21,8 +21,8 @@
     "build": "yarn run build:clean && yarn run build:compile && yarn run build:bundle",
     "build:clean": "rm -rf ./dist && mkdir -p ./dist/esm && mkdir -p ./dist/cjs",
     "build:compile": "yarn build:compile:esm && yarn build:compile:cjs",
-    "build:compile:esm": "tsc --module ES6 --outDir ./dist/esm",
-    "build:compile:cjs": "tsc --module UMD --outDir ./dist/cjs",
+    "build:compile:esm": "tsc --project tsconfig.es2015.json",
+    "build:compile:cjs": "tsc --project tsconfig.json",
     "build:bundle": "browserify --standalone PostMessageStream ./dist/cjs/index.js > ./dist/bundle.js"
   },
   "license": "MIT",
```

With two config files: `tsconfig.json` and `tsconfig.es2015.json`:

```diff
@@ -4,17 +4,15 @@
     "declaration": true,
     "sourceMap": true,
     "strict": true,
-    "target": "es5",
+    "target": "ES5",
+    "module": "CommonJS",
     "lib": [
       "DOM",
       "ES2016"
     ],
-    "outDir": "./dist",
-    "allowJs": true,
-    "allowSyntheticDefaultImports": true
+    "outDir": "./dist/cjs"
   },
   "include": [
-    "./src/**/*.ts",
-    "./src/**/*.js"
+    "./src/**/*.ts"
   ]
-}
\ No newline at end of file
+}
```
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "target": "ES2015",
+    "module": "ES2015"
+  }
+}
```

Reply:
- i think we should build for both (cjs+esm), the ecosystem (node v14, agoric, etc) is trying to get everyone on to esm
- this setup is intended to be a hybrid package https://2ality.com/2019/10/hybrid-npm-packages.html
- umd includes cjs with little overhead, so figured why not. can go either way
- i like moving the config into project files :+1: